crates/history/Cargo.toml

```diff
@@ -30,4 +31,5 @@ gloo-timers = { version = "0.2.6", features = ["futures"], path = "../timers" }
 
 [features]
 query = ["thiserror", "serde_urlencoded"]
+query-qs = ["query", "dep:serde_qs"]
```

Comment:
I am not in favour of including additional serialisers / deserialisers if a decoding / encoding trait can be implemented from outside of `gloo-history`.

Reply:
I am happy to remove it, there is precedent for adding integrations into `serde-qs` itself. 

Although, personally I would advocate for leaving it in for a batteries-included experience. I feel that a lot of people will hit a wall with the default due to [serde_urlencoded not wanting to support sequences](https://github.com/nox/serde_urlencoded/issues/109). 

Even the [urlencoded spec](https://url.spec.whatwg.org/#urlencoded-parsing) supports sequences. They are generated by standards-compliant browsers in HTTP forms linked in the issue. The failure to support these seriously limits the usefulness of the query encoding and decoding for anything but the most basic applications. By shipping with an optional, but more complete strategy, at least we have something to point people at when they run into issues.

Reply:
The primary concern is that if someone else in the future wants a third way of query string especially when one is RFC'ed.
I do not feel there is any more good ground of not accepting such a pull request if we start allowing additional formats in the crate itself.

> Even the [urlencoded spec](https://url.spec.whatwg.org/#urlencoded-parsing) supports sequences. They are generated by standards-compliant browsers in HTTP forms linked in the issue. The failure to support these seriously limits the usefulness of the query encoding and decoding for anything but the most basic applications. 

In `serde_urlencoded`, a sequence can be defined as `Vec<(String, String)>` for arbitrary valid x-form-www-urlencoded sequence, should you need it. `serde_qs` simply do not have a way to produce a sequence without `[]` suffix.

However, because `serde_qs` interprets this differently than the browsers itself, this also limits the usefulness of the implementation.

```html
<form>
    <input name="tag">
    <input name="tag">
    <input type="submit" value="Submit">
</form>
```

Browsers will convert this to `tag=biking&tag=shedding`, however, if I try to use serde_qs to parse the value, it will say: `Custom("Multiple values for one key: \"tag\"")`.

Reply:
It is tricky indeed. 

I understand the possibility to use a `Vec<(String, String)>`, but does that not mean you lose all of the type-safety benefits? Because now you have to manually pull out what key goes where in your struct? 

Possibly an idea would be to go over to that crate and see if we can convince @nox to support sequences, even if it is behind a feature flag. In all of the closed issues, he has indicated that he prefers people use `serde_qs` due to him not wanting to have allocations in the crate (hence why I suggested we build this in here).

 I think I do understand his thinking, but I am quite concerned about having foundational crates like `gloo` or `axum` where basic stuff such as query sequences do not work properly out of the box. It really threw me off when I found out this not to be working in `axum`, and had to awkwardly code around it :/

Will remove the support for `serde_qs` for now so we can have this move ahead. But I will also take a look into the `serde_urlencoded` crate and see if there is something I can do on that side to unbreak this and unblock the people running into this issue.